src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -285,20 +275,21 @@ type TableContext<'TRecord> internal
         if template.PrimaryKey <> rd.PrimaryKey then
             invalidArg (string typeof<'TRecord2>) "incompatible key schema."
 
-        new TableContext<'TRecord2>(client, tableName, rd, defaultMetricsCollector)
+        new TableContext<'TRecord2>(client, tableName, rd, metricsCollector)
 
+    /// Creates a new table context instance that uses a unique metricsCollector
+    member __.WithMetricsCollector(collector : (RequestMetrics -> unit)) : TableContext<'TRecord> =
```

Comment:
should `collector` be optional? (I can't think of a valid usecase tbh)

Reply:
It would only be if you wanted to somehow remove it from a context - I couldn’t think of a good example either...

Reply:
Yeah, marking resolved. I can only think of bad reasons to conceal metrics in special cases given you can trust charges not to be forgotten by the service ;)